server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -449,6 +449,33 @@ public void consumerExists(
         }
     }
 
+    @ApiOperation(
+        notes = "Checks for the existence of a Consumer in bulk. This API return UUIDs of non-existing" +
+        "consumer",
+        value = "")
+    @ApiResponses({ @ApiResponse(code = 404, message = "Returns all consumer UUIDs that doesn't exist or " +
+        "cannot be accessed"),
+        @ApiResponse(code = 204, message = "If the all consumer UUIDs exists and can be accessed"),
+        @ApiResponse(code = 400, message = "When no UUIDs are provided") })
+    @POST
+    @Consumes(MediaType.WILDCARD)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("exists")
+    public Response consumerExistsBulk(Set<String> consumerUuids) throws BadRequestException {
+        if (!CollectionUtils.isEmpty(consumerUuids)) {
+            Set<String> nonExistingUuids = consumerCurator.getNonExistingConsumer(consumerUuids);
```

Comment:
Given the behavior of the getNonExistingConsumers method, this is very risky: you don't know the capabilities of the Set implementation you're passing through.

Reply:
Changed it to `Iterable`  to be more generic.